Dockerfile

```diff
@@ -0,0 +1,101 @@
+FROM ubuntu:16.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && apt-get install --yes --no-install-recommends \
+    automake \
+    bc \
+    bison \
+    ca-certificates \
+    ccache \
+    flex \
+    gawk \
+    gettext \
+    git-core \
+    intltool \
+    liblist-moreutils-perl \
+    liblzo2-dev \
+    libtool \
+    libxml-dom-perl \
+    libxml2-utils \
+    lua5.3 \
+    make \
+    mtd-utils \
+    net-tools \
+    nodejs \
+    pkg-config \
+    qemu-system-arm \
+    qemu-user \
+    rsync \
+    scons \
+    sudo \
+    u-boot-tools \
+    uuid-dev \
+    wget \
+    xutils-dev \
+    xz-utils \
+    zlib1g-dev \
+  && rm --recursive --force /var/lib/apt/lists/* \
+  && ln -sf /usr/bin/lua5.3 /usr/bin/lua \
+  && ln -sf /usr/bin/nodejs /usr/bin/node \
+  && ln -sf /bin/bash /bin/sh
+
+ENV HOME=/home/captain \
+  WSBU_C_COMPILER=/opt/linaro/bin/arm-linux-gnueabihf-gcc \
+  WSBU_CXX_COMPILER=/opt/linaro/bin/arm-linux-gnueabihf-g++ \
+  WSBU_EMULATOR=/usr/bin/qemu-arm \
+  CMAKE_TOOLCHAIN_FILE=/opt/toolchain-linaro-armhf.cmake
```

Comment:
Can you add two more variables, `CC` and `CXX` set to the same values as `WSBU_C_COMPILER` and `WSBU_CXX_COMPILER`? Those are the standard values used by numerous build systems (including Conan) so I would think it would make sense to include those.

Reply:
Actually, we've specifically not exported those because we wanted the build system shim to export only what the dependent needs in its environment. Current mkcommon.inc creates a massive cloud of standardish ENV and nobody knows who all is using what from it.

Reply:
I fixed your other comment about using ENV in toolchain.cmake instead of injecting them via "echo >>" from dockerfile. But I guess since I amended it, the conversation disappeared. Approve @DavidZemon ?